pkg/expr/expr_test.go

```diff
@@ -228,6 +240,111 @@ func TestAssertType(t *testing.T) {
 	}
 }
 
+func TestCacheContent(t *testing.T) {
+	cache, _ := lru.New(1024)
+	cexl := &cexl{
+		fMap: FuncMap(), cache: cache,
+	}
+	str1 := "a.b"
+	_, err := cexl.cacheGetExpression(str1)
+	if err != nil {
+		t.Errorf("Parsing of expression '%s' failed with error '%v'", str1, err)
+		return
+	}
+	expectedEx, _ := Parse(str1)
+
+	exFromCache, _ := cexl.cache.Get(str1)
+	cachedEx := exFromCache.(*Expression)
+
+	if !reflect.DeepEqual(cachedEx, expectedEx) {
+		t.Errorf("Cache contains invalid data: expected=%v, actual=%v", expectedEx, cachedEx)
+	}
+}
+
+func TestCacheHit(t *testing.T) {
+	cache, _ := lru.New(1024)
+	cexl := &cexl{
+		fMap: FuncMap(), cache: cache,
+	}
+	str1 := "a.b"
+	exprNew, err := cexl.cacheGetExpression(str1)
+	if err != nil || exprNew == nil {
+		t.Errorf("Parsing of expression '%s' failed with error '%v'", str1, err)
+		return
+	}
+
+	str2 := "a.b"
+	exprCacheHit, _ := cexl.cacheGetExpression(str2)
+
+	if exprNew != exprCacheHit || cexl.cache.Len() != 1 {
+		t.Errorf("Expected cache hit for expressions '%s' and '%s', and cache size to be 1. Istead cache content is %v.", str1, str2, cexl.cache.Keys())
```

Comment:
Here and below:

Go style is to order messages: got, want. So, I would expect something like: `t.Errorf("No hit for cacheGetExpression(%s); Got cache size of %d with entries %v; expected cache size of 1 with single entry of %s, str2, cexl.cache.Len, cexl.cache.Keys(), str2)`

also: s/Istead/Instead/